test/get-set-delete.test.ts

```diff
@@ -1,221 +1,232 @@
 import {SetupIntegrationTest, isRedisBackedTest} from './integration-setup';
-import {commandOptions} from '@redis/client';
+import {
+  commandOptions,
+  RedisClientType,
+  RedisFunctions,
+  RedisModules,
+  RedisScripts,
+} from '@redis/client';
 import {v4} from 'uuid';
 import {sleep} from './utils';
 
-const {client} = SetupIntegrationTest();
-
-describe('simple get and set', () => {
-  it('should be null on a cache miss', async () => {
-    const key = v4();
-    const result = await client.get(key);
-    expect(result).toBeNull();
-  });
-
-  it('should set and get a string', async () => {
-    const key = v4();
-    const value = v4();
-    const setResult = await client.set(key, value);
-    expect(setResult).toBe('OK');
-    const getResult = await client.get(key);
-    expect(getResult).toEqual(value);
-  });
-
-  it('should allow getting a setting and getting a buffer value', async () => {
-    const key = v4();
-    const value = Buffer.from(v4());
-    const setResult = await client.set(key, value);
-    expect(setResult).toBe('OK');
-    const getResult = await client.get(
-      commandOptions({returnBuffers: true}),
-      key
-    );
-    expect(typeof getResult).toBe('object');
-    expect(getResult).toEqual(value);
+describe('get, set, and delete', () => {
+  let client: RedisClientType<RedisModules, RedisFunctions, RedisScripts>;
+  beforeAll(async () => {
+    client = (await SetupIntegrationTest()).client;
   });
```

Comment:
@pratik151192 @malandis if I change the tests in this lib to use the async factory function to create the momento client, then I need to change how we do the test setup because it's async. I think this is what it will look like.

I will need to do this in a bunch more files to fix the build. Trying to decide if it's worth it or not.

We could theoretically just leave the tests as they were, using the constructor instead of the async factory function, but still use the async factory function in our examples and such. Any opinions?

Reply:
I'm okay with the tests being as is and examples updated; it's very unlikely a customer will snoop in our tests code for samples and/or checking how its wired.

Reply:
okay, reverted that in https://github.com/momentohq/momento-node-redis-client/pull/25/commits/22b108e0922b09a82b60953ecdf507c160bfb872 . I will update the examples after we get a release of this out.